File: 1300.py
from sys import stdin


input = stdin.readline


def solve():
    N = int(input())
    K = int(input())

    lo, hi = 1, K
    while lo <= hi:
        mid = (lo + hi) // 2
        cnt = 0
        for i in range(1, (N if N < mid else mid) + 1):
            val = mid // i
            cnt += val if val < N else N

        if K <= cnt:
            hi = mid - 1
        else:
            lo = mid + 1

    return lo


if __name__ == '__main__':
    print(solve())

File: 1300.js
const fs = require('fs');

function solve() {
	const input = fs.readFileSync('/dev/stdin').toString().split('\n');
	let inputIdx = 0;
	const N = +input[inputIdx++];
	const K = +input[inputIdx++];

	let lo = 1,
		hi = K;
	while (lo <= hi) {
		const mid = Math.floor((lo + hi) / 2);

		let count = 0;
		for (let i = 1; i <= N; i++) {
			const val = Math.floor(mid / i);
			count += val < N ? val : N;
		}
		if (K <= count) hi = mid - 1;
		else lo = mid + 1;
	}
	console.log(lo);
}

solve();
